File: comms/push-notifications/list-push-notification-operations/list-push-notification-operations.php
<?php

namespace Example;

use Twilio\Comms\TwilioComms;
use Twilio\Comms\PushNotifications\Requests\PushNotificationsListOperationsRequest;
use DateTime;
use Twilio\Comms\Types\CommunicationOperationStatus;

$client = new TwilioComms(
    accountId: 'TWILIO_ACCOUNT_SID',
    authToken: 'TWILIO_AUTH_TOKEN',
);
$client->pushNotifications->listOperations(
    new PushNotificationsListOperationsRequest([
        'pageToken' => 'pageToken',
        'pageSize' => 50,
        'startDate' => new DateTime('2024-01-15T09:30:00Z'),
        'endDate' => new DateTime('2024-01-15T09:30:00Z'),
        'status' => CommunicationOperationStatus::Scheduled->value,
    ]),
);

File: comms/push-notifications/list-push-notification-operations/list-push-notification-operations.rb
require "twilio"

client = Twilio::Client.new(
  account_id: "<username>",
  auth_token: "<password>"
)

client.push_notifications.list_operations(
  page_token: "pageToken",
  page_size: 50,
  start_date: "2024-01-15T09:30:00Z",
  end_date: "2024-01-15T09:30:00Z",
  status: "SCHEDULED"
)
